packages/tfchain_client/src/kvstore.ts

```diff
@@ -16,24 +16,60 @@ class KVStore {
     this.client = client;
   }
 
+  /**
+   * Asynchronously sets a `key-value` pair in the `key-value` store.
+   *
+   * This method first checks the connection status, then creates an extrinsic to set the provided `key-value` pair.
+   * It then patches the extrinsic and returns the options that were set.
+   *
+   * @param options - An object containing the key and value to be set in the `key-value` store.
+   * @returns {Promise<ExtrinsicResult<KVStoreSetOptions>>} A promise that resolves to the options that were set in the store.
+   */
   @checkConnection
-  async set(options: KVStoreSetOptions) {
+  async set(options: KVStoreSetOptions): Promise<ExtrinsicResult<KVStoreSetOptions>> {
     const extrinsic = await this.client.api.tx.tfkvStore.set(options.key, options.value);
-    return this.client.patchExtrinsic<void>(extrinsic);
+    return this.client.patchExtrinsic<KVStoreSetOptions>(extrinsic);

```

Comment:
No, it can return an error I think, so I think we need to handle this part. i added the return type here to let who will work on it know that type should be returned